### ISSUE TITLE
onBlur/onFocus don't pass event to asyncValidator

### ISSUE THREAD

**Issue by MiMaMuh** [davidkpiano/react-redux-form#276]
Hi,

I use an custom component created with createFieldClass() anything like this:
(React Native 0.26.3, react-redux-form: 0.13.7)
```js
// just a simple example using createFieldClass() with react-natives' TextInput
const CustomField = createFieldClass({
    'TextInput': (props) => ({
        defaultValue: props.modelValue,
        onChangeText: props.onChange,
        onBlur: props.onBlur,
        onFocus: props.onFocus,
    }),
},{
    componentMap: {
        TextInput: TextInput,
    },
});
```

But when I use a asyncValidator it will not pass any value to the validator function when I use 'blur' or 'focus':
```jsx
// my jsx code using my CustomField component:
<CustomField
    model={'registration.username'}
    asyncValidateOn='blur'
    asyncValidators={{
        usernameAvailable: (val, done) => asyncIsUsernameInUse(val) // <--- val will be undefined??
            .then(inUse => done(!inUse))
            .catch(inUse => done(true)),
    }}
>
    <TextInput />
</CustomField>
```

But it works, when I pass a custom string-value to onBlur for example:
```jsx
<CustomField
    model={'registration.username'}
    asyncValidateOn='blur'
    asyncValidators={{
        usernameAvailable: (val, done) => asyncIsUsernameInUse(val) // <--- val will be 'hello'!!
            .then(inUse => done(!inUse))
            .catch(inUse => done(true)),
    }}
>
    <TextInput 
       onBlur={() => this.props.onBlur && this.props.onBlur('hello')} // <--- this will work!
    />
</CustomField>
```

When I pass the event, it won't work:
```jsx
<CustomField
    model={'registration.username'}
    asyncValidateOn='blur'
    asyncValidators={{
        usernameAvailable: (val, done) => asyncIsUsernameInUse(val) // <--- val will be undefined again!
            .then(inUse => done(!inUse))
            .catch(inUse => done(true)),
    }}
>
    <TextInput 
       onBlur={(event) => this.props.onBlur && this.props.onBlur(event)} // <--- now I pass the event manually
    />
</CustomField>
```

Is it a bug? Or how can I get the value?

**Comment by davidkpiano** [comment]
@MiMaMuh This doesn't seem to be occurring in `0.14.2`. Can you please confirm here by looking at this test case (which is pretty much a copy-paste of your implementation)?

https://github.com/davidkpiano/react-redux-form/commit/6d31d97ebc0a7c00c3fff7304d2ebac673f8dd15#diff-ded95f7363bb42a57e08873c85c00260R234

**Comment by MiMaMuh** [comment]
@davidkpiano Hi, yes, your test-case looks fine. I just looked to my issue: I updated to react-redux-form 0.14.1 and tried it again.

I stumbled about a new issue with my temporary fix. To bridge this issue I wrote a ```onBlur``` function with: 
```jsx
<TextInput 
  onBlur={(event) => this.props.onBlur && this.props.onBlur(event.nativeEvent.text)}
/>
```

This works just fine with ```asyncValidateOn={'change'}``` but I just figured out it is broken with ```asyncValidateOn={'blur'}```. It seems with react-native (0.26.3) the ```event.nativeEvent``` doesn't get the text property. 

It may be a bug in react-native as i found this [issue from Dec 2005](https://productpains.com/post/react-native/android-textinput-the-event-argument-passed-in-onblur-function-doesnt-contains-correct-nativeevent-property) – I'm not sure if this fits to my issue as I tested it on iOS, but it looks similar.

Currently I think it isn't an issue of react-redux-form as your test seems fine. I haven't the time to upgrade my project to the latest react-native version and test it again, but I'll post it here when it seems to be fixed.

Thank you for your help!
